### PR TITLE
Fixes #19177: Improve linter and test configuration

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/19177

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: qa-test [hunk 1]
```diff
@@ -2,37 +2,81 @@
...
   exit 0
 elif [ "$1" = "--scala" ]; then
-  cd webapp/sources && mvn clean install -Dmaven.test.postgres=false
+  test_webapp_hooks
+  test_webapp_mvn
+  exit 0
+elif [ "$1" = "--shell" ]; then
+  test_shell
   exit 0
+else
```
Comment (peckpeck) [inline qa-test:76]: merge already passes --merge as an argument

Comment (amousset) [reply qa-test:76]: do we want a different behavior when no arguments are passed?

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: qa-test (hunk 1 above)
Comment (peckpeck) [reply qa-test:38]: i would have said run the longest tests

**Review by VinceMacBuche — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request closed without merging**